cmd/virt-handler/virt-handler.go

```diff
@@ -73,7 +74,13 @@ func main() {
 	broadcaster.StartRecordingToSink(&kubecorev1.EventSinkImpl{Interface: coreClient.Events(api.NamespaceDefault)})
 	recorder := broadcaster.NewRecorder(kubev1.EventSource{Component: "virt-handler", Host: *host})
 
-	domainManager, err := virtwrap.NewLibvirtDomainManager(domainConn, recorder)
+	var domainManager virtwrap.DomainManager
+	switch *driver {
+	case "ovirt":
+		domainManager, err = virtwrap.NewOvirtDomainManager(domainConn, recorder)
+	default:
+		domainManager, err = virtwrap.NewLibvirtDomainManager(domainConn, recorder)
+	}
```

Comment:
I don't think we have came to such conclusion yet.

That being said, interaction through k8s API is preferable yet may not be easily done without requiring breaking changes to oVirt. We need to figure out how to enable coexistence of VDSM and virt-handler to see how these projects could benefit from each other.

I myself would love to see some domain manager extension endpoint that would allow users to use oVirt, libvirt or any other virtualization provider. That would bring us philosophically close to k8s.